### PR TITLE
[SPARK-13358][SQL] Retrieve grep path when do benchmark

### PR DESCRIPTION
**Pull request by viirya**
JIRA: https://issues.apache.org/jira/browse/SPARK-13358

When trying to run a benchmark, I found that on my Ubuntu linux grep is not in /usr/bin/ but /bin/. So wondering if it is better to use which to retrieve grep path.

cc @davies

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/util/Benchmark.scala [hunk 1]
```diff
@@ -93,7 +93,8 @@ private[spark] object Benchmark {
     if (SystemUtils.IS_OS_MAC_OSX) {
       Utils.executeAndGetOutput(Seq("/usr/sbin/sysctl", "-n", "machdep.cpu.brand_string"))
     } else if (SystemUtils.IS_OS_LINUX) {
-      Utils.executeAndGetOutput(Seq("/usr/bin/grep", "-m", "1", "\"model name\"", "/proc/cpuinfo"))
+      val grepPath = Utils.executeAndGetOutput(Seq("which", "grep"))
+      Utils.executeAndGetOutput(Seq(grepPath, "-m", "1", "model name", "/proc/cpuinfo"))
```
Comment (viirya) [inline core/src/main/scala/org/apache/spark/util/Benchmark.scala]: "\"model name\"" doesn't work on my linux. "model name" works. Don't know if it is special case or general.

**Comment by viirya** [comment]
retest this please.

File: core/src/main/scala/org/apache/spark/util/Benchmark.scala [hunk 2]
```diff
@@ -93,7 +93,8 @@ private[spark] object Benchmark {
     if (SystemUtils.IS_OS_MAC_OSX) {
       Utils.executeAndGetOutput(Seq("/usr/sbin/sysctl", "-n", "machdep.cpu.brand_string"))
     } else if (SystemUtils.IS_OS_LINUX) {
-      Utils.executeAndGetOutput(Seq("/usr/bin/grep", "-m", "1", "\"model name\"", "/proc/cpuinfo"))
```
Comment (viirya) [inline core/src/main/scala/org/apache/spark/util/Benchmark.scala]: Or just make it "grep" without specified path?

**Comment by hvanhovell** [comment]
I guess there isn't a way to test this... Lemme try this locally.

**Comment by hvanhovell** [comment]
This works on my linux machine.

File: core/src/main/scala/org/apache/spark/util/Benchmark.scala (hunk 1 above)
Comment (hvanhovell) [inline core/src/main/scala/org/apache/spark/util/Benchmark.scala]: Same here.

**Comment by hvanhovell** [comment]
I guess the whole mechanism is a bit brittle, shall we add a somekind of error handling (like in #11209) so Benchmarking won't fail just because it is done on some unknown architecture.

@nongli (nice talk today BTW) You added this. On which linux platform did you test this?

**Comment by viirya** [comment]
ping @davies do you think this is useful? Thanks.

**Comment by davies** [comment]
LGTM, merging into master, thanks!

**Pull request closed without merging**